microbench/src/main/java/io/netty/handler/codec/http/HttpMethodMapBenchmark.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static io.netty.handler.codec.http.HttpMethod.CONNECT;
+import static io.netty.handler.codec.http.HttpMethod.DELETE;
+import static io.netty.handler.codec.http.HttpMethod.GET;
+import static io.netty.handler.codec.http.HttpMethod.HEAD;
+import static io.netty.handler.codec.http.HttpMethod.OPTIONS;
+import static io.netty.handler.codec.http.HttpMethod.PATCH;
+import static io.netty.handler.codec.http.HttpMethod.POST;
+import static io.netty.handler.codec.http.HttpMethod.PUT;
+import static io.netty.handler.codec.http.HttpMethod.TRACE;
+import static io.netty.util.internal.MathUtil.findNextPositivePowerOfTwo;
+
+@State(Scope.Benchmark)
+@Warmup(iterations = 5)
+@Measurement(iterations = 8)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+public class HttpMethodMapBenchmark extends AbstractMicrobenchmark {
+    private static final Map<String, HttpMethod> oldMap = new HashMap<String, HttpMethod>();
+    private static final SimpleStringMap<HttpMethod> newMap;
+    private static final String[] knownMethods;
+    private static final String[] mixMethods;
+    private static final String[] unknownMethods;
+
+    static {
+        knownMethods = new String[] {
+                "OPTIONS",
+                "GET",
+                "HEAD",
+                "POST",
+                "PUT",
+                "PATCH",
+                "DELETE",
+                "TRACE",
+                "CONNECT"
```

Comment:
consider using `HttpMethod.OPTIONS.getName()` etc

Reply:
It was this way originally but this doesn't simulate how the HttpObjectDecoder actually works (it builds a new String object), and was unfairly biased toward the new approach due to the `equals` short circuiting on reference equality checks.

I will add comments.

Reply:
ok